ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackendFactory.scala

```diff
@@ -5,9 +5,20 @@ package com.daml.platform.store.backend.common
 
 import com.daml.platform.store.backend._
 import com.daml.platform.store.cache.LedgerEndCache
+import com.daml.platform.store.interning.StringInterning
 
 trait CommonStorageBackendFactory extends StorageBackendFactory {
 
+  protected def queryStrategy: QueryStrategy
```

Comment:
May I ask you not to add template method to this trait? I understand your motivation, but the code which you remove is not gigantic, and direct dependency injection is much cleaner.
Also: could you pls move pure refactorings to different PRs if possible? Makes PRs much more focused, and easier to review.